### PR TITLE
Create models for Question, Choice and Answer

### PR DESCRIPTION
**Pull request by kingdomlevel**
This PR:
- uses an app called `assessment`
- Creates a model for the question database table - (_ticket 411_)
- Creates a model for the choices database table - (_ticket 412_)
- Creates a model for the answers database table - (_ticket 416_)
- Creates unit tests in `tests/test_models` - (_ticket 413_)
- Create a migration to new table to store the content of the question (_ticket 408_)
- Create a migration to new table to store the content of the possible choices (_ticket 409_)
- Create a migration to new table to store the selected answer to a question (_ticket 415_)
- ~~creates a sample question "on the fly" and renders this on a sample index page~~
- renders index page from dynamic pages app
- ~~suppresses~~ resolves no root directory warning when running tests before `collectstatic` is run

### CODE REVIEW

File: CyberHealth/CyberHealth/urls.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 urlpatterns = [
     path('admin/', admin.site.urls),
+    path('assessment/', include('assessment.urls')),
     path('dynamic/', include('dynamicpages.urls')),
```
Comment (joel-stobart) [inline CyberHealth/CyberHealth/urls.py:22]: We can probably drop the dynamic app at this point?

**Review by joel-stobart — CHANGES_REQUESTED** [review]

**Review by h0usemaintenance — CHANGES_REQUESTED** [review]
- If the dynamicpages app is not required please remove it.
- Please provide evidence of a full working travis build
- Please amend the model test class name as you seem to be testing all your models in that same class. Shouldn't it be separate test .py files as they are separate models?
- Shouldn't the index page be reserved for static content as demonstrated by recent prototypes?

**Comment by kingdomlevel** [comment]
Quoting the above in order to track progress:
- [x] If the dynamicpages app is not required please remove it.
- [ ] Please provide evidence of a full working travis build
- [ ] Please amend the model test class name as you seem to be testing all your models in that same class. Shouldn't it be separate test .py files as they are separate models?
- [ ] classes should have constructors
- [ ] Shouldn't the index page be reserved for static content as demonstrated by recent prototypes?
- Please use the project repository for development rather than fork from your github workspace.
    - not applicable

**Comment by h0usemaintenance** [comment]
Can you rollback your changes to the Travis file please? and confirm that all functionality for dynamicpages was moved to the new app assessment?

**Review by h0usemaintenance — CHANGES_REQUESTED** [review]
- There are no changes to the models.py from previous pull request
- assessment/urls please give the view a more relevant name based on what you are trying to achieve
- No information added to any of the jiras

**Comment by kingdomlevel** [comment]
> * There are no changes to the models.py from previous pull request
> * assessment/urls please give the view a more relevant name based on what you are trying to achieve

Discussed the above on a call at the end of the day today - this PR is to be left for Joel to review tomorrow after he gets back from his fun birthday day off! 🥳

**Comment by h0usemaintenance** [comment]
- Can you provide some evidence in relation to the relationships between the tables?
- I found the answers I needed re constructors in Django.

**Comment by kingdomlevel** [comment]
> * Can you provide some evidence in relation to the relationships between the tables?
from dbshell:
```
sqlite> .schema assessment_question 
CREATE TABLE IF NOT EXISTS "assessment_question" ("id" integer NOT NULL PRIMARY KEY AUTOINCREMENT, "question_text" varchar(255) NOT NULL);
sqlite> .schema assessment_choice
CREATE TABLE IF NOT EXISTS "assessment_choice" ("id" integer NOT NULL PRIMARY KEY AUTOINCREMENT, "choice_text" varchar(255) NOT NULL, "question_id" integer NOT NULL REFERENCES "assessment_question" ("id") DEFERRABLE INITIALLY DEFERRED);
CREATE INDEX "assessment_choice_question_id_2d5f9dae" ON "assessment_choice" ("question_id");
sqlite> .schema assessment_answer
CREATE TABLE IF NOT EXISTS "assessment_answer" ("id" integer NOT NULL PRIMARY KEY AUTOINCREMENT, "choice_id" integer NOT NULL REFERENCES "assessment_choice" ("id") DEFERRABLE INITIALLY DEFERRED, "question_id" integer NOT NULL REFERENCES "assessment_question" ("id") DEFERRABLE INITIALLY DEFERRED);
CREATE INDEX "assessment_answer_choice_id_17c2075a" ON "assessment_answer" ("choice_id");
CREATE INDEX "assessment_answer_question_id_0cabff66" ON "assessment_answer" ("question_id");
```
+screenshot: 
image

See also [models](CyberHealth/assessment/models.py) and the models tests for the models with foreign keys: [choice](CyberHealth/assessment/tests/models/test_choice.py), [answer](CyberHealth/assessment/tests/models/test_answer.py) 

> * creates a sample question "on the fly" and renders this on a sample index page - where is this?
legacy, was true as of [when the PR was  opened](https://github.com/WeAreSnook/cyber-health-frontend/pull/29/commits/ad3988cd684de5018b69c0e24a932382d8485376), outdated since [4106e6f](https://github.com/WeAreSnook/cyber-health-frontend/pull/29/commits/4106e6fba0de8de00e57404f7b959b1274812f93). Will ammend initial PR comment.

> * suppresses no root directory warning when running tests before collectstatic is run - why suppress it rather than resolve it?
     - language issue, this is resolved. will amend initial PR comment.

**Review by h0usemaintenance — CHANGES_REQUESTED** [review]
- Please correct file CyberHealth/staticpages/templates/staticpages/index.html as Joel has amended that code(also on main)

**Comment by kingdomlevel** [comment]
@h0usemaintenance  - this passed in [build #447](https://travis-ci.org/github/WeAreSnook/cyber-health-frontend/builds/767177425)!

**Review by h0usemaintenance — CHANGES_REQUESTED** [review]
Cannot be merged yet as there is some refactor to do & models need to be tested with recent changes to main.

**Review by joel-stobart — APPROVED** [review]
I am happy with these changes

**Review by h0usemaintenance — APPROVED** [review]
Change looks as expected.

**Pull request merged**